### ISSUE TITLE
Calculating exposure

### ISSUE THREAD

**Issue by wibeasley** [OuhscBbmc/referral-exposure#6]
@DavidBard, I think this function is ready to discuss, and how it should behave with some of the corner cases, particularly when the exposure is non-continuous.

Here's the [testing code](https://github.com/OuhscBbmc/referral-exposure/blob/master/tests/testthat/test-exposure.R) that specifies the desired output for the following input.  Here's [the actual function](https://github.com/OuhscBbmc/referral-exposure/blob/master/R/exposure.R).  Notice that the bottom of the test file looks at each output column separately.  This anticipates that we might wander into whackamole territory, and accidentally create problems while fixing others.

```r
ds_kid_referral_input <-
  tibble::tribble(
    ~client_id, ~referral_id, ~referral_date, ~was_removed, ~removal_begin_date, ~was_returned, ~removal_end_date,
            1L,           1L,   "2014-01-01",        FALSE,       NA_character_,         FALSE,     NA_character_,
            1L,           1L,   "2014-01-01",         TRUE,        "2014-02-01",          TRUE,      "2014-04-01",
            1L,           2L,   "2015-01-01",         TRUE,        "2015-02-01",          TRUE,      "2015-06-01",
            1L,           3L,   "2016-01-01",        FALSE,       NA_character_,         FALSE,     NA_character_,

            2L,           4L,   "2014-01-01",        FALSE,       NA_character_,         FALSE,     NA_character_,
            2L,           5L,   "2014-07-01",         TRUE,        "2014-08-01",          TRUE,      "2014-10-01",
            2L,           6L,   "2015-01-01",         TRUE,        "2015-03-01",         FALSE,     NA_character_,

            3L,           7L,   "2014-01-01",        FALSE,       NA_character_,         FALSE,     NA_character_,
            3L,           8L,   "2015-01-01",        FALSE,       NA_character_,         FALSE,     NA_character_,

            4L,           9L,   "2015-01-01",        FALSE,       NA_character_,         FALSE,     NA_character_,

            5L,          10L,   "2014-08-01",         TRUE,        "2014-11-01",          TRUE,      "2014-12-01"
  )
```

**Comment by wibeasley** [comment]
@DavidBard bump

**Comment by DavidBard** [comment]
@wibeasley This is exciting and looks very useful!  My only concern is that the two most interesting variables "referral_date_subsequent" and "removal_begin_date_subsequent" get commented out in the very last code block.  We'll need those to construct the time to subsequent referrals and removals, and we'll want to carefully consider the use of calipers to ignore any subsequent referrals that happen too close in time to the initial referral (they are likely referrals for the same concerning safety threat(s) just made by a separate individual, e.g., a teacher one week, then a neighbor the next).  

One other complication that this code reminded me of, is that the concept of being "returned" is trickier than it might sound.  I believe your was_returned variable only considers whether or not the children are reunified with the PRFCs who were providing care for the child at the time of a referral.  During these subsequent referrals and removals, we may be dealing with different PRFCs for kids (e.g., kid gets removed from PRFC 1 at initial referral, is adopted by PRFC 2, and then a year later a new referral surfaces with an allegation involving PRFC 2).  For purposes of the waiver, I don't think DHS cares as much about the switching of PRFCs.  They are mostly concerned with how long will the child be under the custody of the State.  For this reason, I think we also need a "was_released" variable (in addition to "was_returned" which is important to track) to capture any time a child leaves State custody.  Thoughts?

**Comment by wibeasley** [comment]
yeah, i started to appreciate the complications caused be overlapping ~~segments~~ spans.  I was thinking mostly about referrals overlapping each other, but you're right about the kids and prfcs not having clean one-to-one boundaries either.

It makes something like #8 harder than I originally thought.  But probably more important too.
